lib/vue/src/components/map/CMap.vue

```diff
@@ -1,57 +1,84 @@
-  @update:bounds="onBoundsUpdate"
->
-  <l-tile-layer
-    url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
-    :attribution="map.attribution"
-  />
-
-  <l-polygon
-    v-for="(geometry, i) in polygonGeometries"
-    :key="`polygon-${i}`"
-    :lat-lngs="geometry.map(value => value.geometry)"
-    :color="polygonColors[i]"
-  />
-
-  <l-marker
-    v-for="(marker, i) in markers"
-    :key="i"
-    :lat-lng="marker"
-    :opacity="localValueIndex === undefined || i == localValueIndex ? 1.0 : 0.6"
-    @click="onRemoveMarker(i)"
-  />
-
-  <slot name="markers"></slot>
-
-  <l-control class="leaflet-bar">
-    <a
-      v-if="!hideCurrentLocationButton"
-      :title="label.tooltip && label.tooltip.goToCurrentLocation"
-      role="button"
-      class="d-flex justify-content-center align-items-center"
-      @click="onGoToCurrentLocation"
+  <div class="position-relative" v-if="mapOptions.show">
+    <div
+      v-if="!hideGeoSearch"
+      class="geosearch-container"
+      @mouseover="disableMap"
+      @mouseleave="enableMap"
     >
-      <font-awesome-icon
-        :icon="['fas', 'location-arrow']"
-        class="text-primary"
+      <c-input-search
+        v-model="geoSearch.query"
+        :placeholder="labels.geosearchInputPlaceholder"
+        :autocomplete="'off'"
+        :debounce="300"
+        @input="onGeoSearch"
       />
-    </a>
-  </l-control>
-</l-map>
+
+      <div class="geosearch-results">
+        <div
+          v-for="(result, idx) in geoSearch.results"
+          :key="idx"
+          class="geosearch-result"
+          @click="placeGeoSearchMarker(result)"
+        >
+          {{ result.label }}
+        </div>
+      </div>
+    </div>
+
+    <l-map
+      ref="map"
+      :zoom="mapOptions.zoom"
+      :center="mapOptions.center"
+      :min-zoom="mapOptions.minZoom"
+      :max-zoom="mapOptions.maxZoom"
+      :bounds="mapOptions.bounds"
+      :max-bounds="mapOptions.maxBounds"
+      style="cursor: pointer;"
+      :style="mapStyle"
+      @click="onMapClick"
+      @locationfound="onLocationFound"
+      @update:zoom="onZoom"
+      @update:center="onCenter"
+      @update:bounds="onBoundsUpdate"
+    >
+      <l-tile-layer
+        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+        :attribution="mapOptions.attribution"
+      />
+    
+      <l-polygon
+        v-for="(geometry, i) in polygons"
+        :key="i"
+        :lat-lngs="geometry.map(value => value.geometry)"
+        :color="geometry[0].color"
+      />
+    
+      <l-marker
+        v-for="(marker, i) in markers"
+        :key="`marker-${i}`"
+        :lat-lng="marker"
+        :icon="getIcon(marker.color)"
+        :opacity="activeMarkerIndex === undefined || i == activeMarkerIndex ? 1.0 : 0.6"
+        @click="onMarkerClick(i, marker)"
+      />
+    
+      <l-control class="leaflet-bar">
+        <a
+          v-if="!hideCurrentLocationButton"
+          :title="labels.tooltip && labels.tooltip.goToCurrentLocation"
```

Comment:
Probably worth it to use a tooltip here so you don't have to do it on the other issue